test/testpaint/CMakeLists.txt

```diff
@@ -19,8 +19,14 @@ file(GLOB_RECURSE ORCT2_TESTPAINT_SOURCES "${CMAKE_CURRENT_LIST_DIR}/*.c"
                                           "${CMAKE_CURRENT_LIST_DIR}/*.cpp"
                                           "${CMAKE_CURRENT_LIST_DIR}/*.h")
 
-add_executable(testpaint EXCLUDE_FROM_ALL ${ORCT2_RIDE_SOURCES} ${ORCT2_RIDE_DEP_SOURCES} ${ORCT2_TESTPAINT_SOURCES} ${RCT2_SECTIONS})
-target_include_directories(testpaint PRIVATE "${ROOT_DIR}/src/")
-set_target_properties(testpaint PROPERTIES COMPILE_FLAGS "-DNO_VEHICLES -D__TESTPAINT__ -Wno-unused")
-set_target_properties(testpaint PROPERTIES LINK_FLAGS ${RCT2_SEGMENT_LINKER_FLAGS})
-add_dependencies(testpaint segfiles)
+add_executable(testpaint
+    ${ORCT2_RIDE_SOURCES}
+    ${ORCT2_RIDE_DEP_SOURCES}
+    ${ORCT2_TESTPAINT_SOURCES}
+    )
+target_include_directories(testpaint PRIVATE "${PROJECT_SOURCE_DIR}/src/")
+target_compile_definitions(testpaint PRIVATE NO_VEHICLES __TESTPAINT__)
+target_compile_options(testpaint PRIVATE -Wno-unused -fno-omit-frame-pointer)
+target_link_libraries(testpaint PRIVATE openrct2::sections openrct2::flags)
+add_test(NAME testpaint COMMAND testpaint --quiet)
+set_property(TEST testpaint PROPERTY WILL_FAIL TRUE)
```

Comment:
The lines above are a bit dark magic, it would be nice to have them commented a little.
* `-fno-omit-frame-pointer` is required for optimised builds to work with RCT2 interop enabled
* `--quiet` is to limit the verbosity of `testpaint`, otherwise it overflows Travis' logs
* `WILL_FAIL` is on, because there are known differences in drawing code. @marijnvdwerf is there an issue about that or other link we could provide for status/explanation?

Reply:
@janisozaur I don't think so. Might be good to create one though.

Reply:
ping @marijnvdwerf